File: iottalk_server_1.0/da/Map/FetchData/FetchData_Airbox/csmapi.py
import requests

ENDPOINT = None
TIMEOUT=10
IoTtalk = requests.Session()
#print(IoTtalk)

class CSMError(Exception):
    pass

def register(mac_addr, profile, UsingSession=IoTtalk):
    r = UsingSession.post(
        ENDPOINT + '/' + mac_addr,
        json={'profile': profile}, timeout=TIMEOUT
    )
    if r.status_code != 200: raise CSMError(r.text)
    return True


def deregister(mac_addr, UsingSession=IoTtalk):
    r = UsingSession.delete(ENDPOINT + '/' + mac_addr)
    if r.status_code != 200: raise CSMError(r.text)
    return True


def push(mac_addr, df_name, data, UsingSession=IoTtalk):
    r = UsingSession.put(
        ENDPOINT + '/' + mac_addr + '/' + df_name,
        json={'data': data}, timeout=TIMEOUT
    )
    if r.status_code != 200: raise CSMError(r.text)
    return True


def pull(mac_addr, df_name, EP=ENDPOINT, UsingSession=IoTtalk):
    r = UsingSession.get(EP + '/' + mac_addr + '/' + df_name, timeout=TIMEOUT)
    if r.status_code != 200: raise CSMError(r.text)
    return r.json()['samples']


def get_alias(mac_addr, df_name, UsingSession=IoTtalk):
    r = UsingSession.get(ENDPOINT + '/get_alias/' + mac_addr + '/' + df_name, timeout=TIMEOUT)
    if r.status_code != 200: raise CSMError(r.text)
    return r.json()['alias_name']

	
def set_alias(mac_addr, df_name, s, UsingSession=IoTtalk):
    r = UsingSession.get(ENDPOINT + '/set_alias/' + mac_addr + '/' + df_name + '/alias?name=' + s, timeout=TIMEOUT)
    if r.status_code != 200: raise CSMError(r.text)
    return True

	
def tree(UsingSession=IoTtalk):
    r = UsingSession.get(ENDPOINT + '/tree')
    if r.status_code != 200: raise CSMError(r.text)
    return r.json()

	
def pull_airbox(url, UsingSession=IoTtalk):
    r = UsingSession.get(url, verify=False, timeout=TIMEOUT)
    if r.status_code != 200: raise CSMError(r.text)
    return r.json()['devices']

File: iottalk_server_1.0/da/Graph/js/csmapi.js
var csmapi = (function () {
    var ENDPOINT = window.location.origin;

    function set_endpoint (endpoint) {
        ENDPOINT = endpoint;
    }

    function get_endpoint () {
        return ENDPOINT;
    }

    function register (mac_addr, profile, callback) {
        jQuery.ajax({
            type: 'POST',
            url: ENDPOINT +'/'+ mac_addr,
            data: JSON.stringify({'profile': profile}),
            contentType:"application/json; charset=utf-8",
        }).done(function () {
            if (callback) {
                callback(true);
            }
        }).fail(function () {
            if (callback) {
                callback(false);
            }
        });
    }

    function deregister (mac_addr, callback) {
        jQuery.ajax({
            type: 'DELETE',
            url: ENDPOINT +'/'+ mac_addr,
            contentType:"application/json; charset=utf-8",
        }).done(function () {
            if (callback) {
                callback(true);
            }
        }).fail(function () {
            if (callback) {
                callback(false);
            }
        });
    }

    function pull (mac_addr, odf_name, callback) {
        jQuery.ajax({
            type: 'GET',
            url: ENDPOINT +'/'+ mac_addr +'/'+ odf_name,
            contentType:"application/json; charset=utf-8",
        }).done(function (obj) {
            if (typeof obj === 'string') {
                obj = JSON.parse(obj);
            }

            if (callback) {
                callback(obj['samples']);
            }
        }).fail(function (error) {
            if (callback) {
                callback([], error);
            }
        });
    }

    function get_alias_name (mac_addr, odf_name, callback) {
        jQuery.ajax({
            type: 'GET',
            url: ENDPOINT +'/get_alias/'+ mac_addr +'/'+ odf_name,
            contentType:"application/json; charset=utf-8",
        }).done(function (obj) {
            if (typeof obj === 'string') {
                obj = JSON.parse(obj);
            }

            if (callback) {
                callback({alias: obj['alias_name'], origin: odf_name});
            }
        }).fail(function (error) {
            if (callback) {
                callback([], error);
            }
        });
    }

    function push (mac_addr, idf_name, data, callback) {
        jQuery.ajax({
            type: 'PUT',
            url: ENDPOINT +'/'+ mac_addr +'/'+ idf_name,
            data: JSON.stringify({'data': data}),
            contentType:"application/json; charset=utf-8",
        }).done(function () {
            if (callback) {
                callback(true);
            }
        }).fail(function () {
            if (callback) {
                callback(false);
            }
        });
    }

    return {
        'set_endpoint': set_endpoint,
        'get_endpoint': get_endpoint,
        'register': register,
        'deregister': deregister,
        'pull': pull,
        'get_alias_name': get_alias_name,
        'push': push,
    };
})();